### PR TITLE
Bugfix9

### PR DESCRIPTION
**Pull request by Kredcool**
this attempts to fix the not being able to go over 360

### CODE REVIEW

File: dashboard/.python-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.8.0
```
Comment (KenwoodFox) [inline dashboard/.python-version]: Dont include this

File: docs/.python-version [hunk 2]
```diff
@@ -0,0 +1 @@
+3.8.0
```
Comment (KenwoodFox) [inline docs/.python-version]: Dont include this

File: rio/subsystems/drivetrain.py [hunk 3]
```diff
@@ -105,68 +120,141 @@ def configureNetworkTables(self):
...
+            optimizedState = kinematics.SwerveModuleState.optimize(
+                newState, self.getState().angle
+            )
+
+            # Get the desired position (in neo rotations) given by the optimized module state
+            currentRef = (optimizedState.angle.radians() / (2 * math.pi)) * self.pid[
+                "steer"
+            ]["ratio"]
+
+            # this checks if the desiredState is more than 349 and currentRef is 59 then adds 2 pi to currentRef
```
Comment (KenwoodFox) [inline rio/subsystems/drivetrain.py]: Degrees and radians dont mix

File: rio/subsystems/drivetrain.py [hunk 4]
```diff
@@ -105,68 +120,141 @@ def configureNetworkTables(self):
...
+            if (
```
Comment (KenwoodFox) [inline rio/subsystems/drivetrain.py]: This is pretty messy

File: rio/subsystems/drivetrain.py [hunk 5]
```diff
@@ -105,68 +120,141 @@ def configureNetworkTables(self):
...
+                self.desiredState >= currentRef
+                and self.desiredState.angle.radians() >= 6.3490912
+                and currentRef.angle.radians() <= 1.0297
+            ):
+                currentRef = currentRef.angle.radians() + math.pi + math.pi
```
Comment (KenwoodFox) [inline rio/subsystems/drivetrain.py]: This is an odd way of adding 2pi but i see what you're trying to do

**Review by KenwoodFox — CHANGES_REQUESTED** [review]

File: rio/subsystems/drivetrain.py (hunk 5 above)
Comment (Kredcool) [reply rio/subsystems/drivetrain.py]:
```python
2 * math.pi```

**Pull request merged**